SynthDet/Assets/Scripts/Runtime/SynthDetScenario.cs

```diff
@@ -19,142 +20,102 @@ namespace SynthDet.Scenarios
     {
-        int m_NumPrefabsToLoad;
-        List<GameObject> m_Prefabs = new List<GameObject>();
-        AssetLoadingStatus m_LoadingStatus = AssetLoadingStatus.LoadingCatalog;
-        AsyncOperationHandle<IResourceLocator>[] m_CatalogHandles;
-
+        static HashSet<Color> s_ColorsAlreadyUsed = new HashSet<Color>();
+        static ColorRgbParameter s_SemanticColorParameter = new ColorRgbParameter();
+        
+        AssetLoadingStatus m_LoadingStatus = AssetLoadingStatus.InProgress;
         List<string> m_LabelStringsForAutoLabelConfig = new List<string>();
 
         /// <inheritdoc/>
         protected override bool isScenarioReadyToStart
         {
             get
             {
-                switch (m_LoadingStatus)
-                {
-                    case AssetLoadingStatus.Complete:
-                        return true;
-                    case AssetLoadingStatus.LoadingCatalog:
-                    {
-                        foreach (var handle in m_CatalogHandles)
-                        {
-                            if (!handle.IsDone)
-                                return false;
-                        }
-                        
-                        foreach (var handle in m_CatalogHandles)
-                        {
-                            if (handle.Status != AsyncOperationStatus.Succeeded)
-                            {
-                                Debug.LogError("Catalog download failed");
-                                m_LoadingStatus = AssetLoadingStatus.Failed;
-                                return false;
-                            }
-                        }
-                        
-                        LoadAndLabelPrefabs();
-                        m_LoadingStatus = AssetLoadingStatus.LoadingPrefabs;
-                        return false;
-                    }
-                    case AssetLoadingStatus.LoadingPrefabs:
-                    {
-                        lock (m_Prefabs)
-                        {
-                            if (m_Prefabs.Count < m_NumPrefabsToLoad)
-                                return false;
-                            SetupLabelConfigs();
-                            var randomizer = GetRandomizer<ForegroundObjectPlacementRandomizer>();
-                            m_Prefabs.Sort((prefab1, prefab2) => prefab1.name.CompareTo(prefab2.name));
-                            randomizer.prefabs = m_Prefabs.ToArray();
-                            m_LoadingStatus = AssetLoadingStatus.Complete;
-                            return true;
-                        }
-                    }
-                    case AssetLoadingStatus.Failed:
-                    {
-                        Quit();
-                        return false;
-                    }
-                    default:
-                        return false;
-                }
+                if (m_LoadingStatus == AssetLoadingStatus.Complete)
```

Comment:
Out of curiosity, why can't we completely block the main thread until all of the loading is complete?

Reply:
I think I ran into some problems with waiting on the async Task returned from addressables methods, but I'll experiment with this again to see if we can just block the main thread like you suggested.

Reply:
Alrighty, the whole loading process should block the main thread now.